src/plugins/controls/public/control_group/editor/edit_control.tsx

```diff
@@ -22,19 +22,31 @@ import { IEditableControlFactory, ControlInput } from '../../types';
 import { controlGroupReducers } from '../state/control_group_reducers';
 import { ControlGroupContainer, setFlyoutRef } from '../embeddable/control_group_container';
 
+interface EditControlResult {
+  type: string;
+  controlInput: Omit<ControlInput, 'id'>;
+}
+
 export const EditControlButton = ({ embeddableId }: { embeddableId: string }) => {
   // Controls Services Context
   const { overlays, controls } = pluginServices.getHooks();
   const { getControlFactory } = controls.useService();
   const { openFlyout, openConfirm } = overlays.useService();
 
+  let promiseResolve: (result: EditControlResult) => void;
+  let promiseReject: () => void;
+  const initialInputPromise = new Promise<EditControlResult>((resolve, reject) => {
+    promiseResolve = resolve;
+    promiseReject = reject;
+  });
+
```

Comment:
Just like `createControl`, this is now a promise since this should happen `async` - specifically, we'll want to wait for the new embeddable to load before resolving.